### ISSUE TITLE
Apple Silicon m1 version 

### ISSUE THREAD

**Issue by paoloestar** [prashantgupta24/automatic-mouse-mover#33]
Hey prashanta, very good and efficient piece of software thanks a lot!

have seen (after buying a new MacBook) that it still runs rosetta emulated as intel application. Any plans to make a universal app?  Thanks!

**Comment by prashantgupta24** [comment]
Thanks for the kind words @paoloestar! No plans as of now, contributions are more than welcome!

**Comment by longgiangp** [comment]
With the release of Go, [version 1.16] Apple Silicon is supported out-of-the-box with support for cgo, internal and external linking. https://tip.golang.org/doc/go1.16

By looking at your go.yml file, you're already on Go 1.17. which means you should be able to build for arm (GOOS=darwin, GOARCH=arm64). 

If you don't mind to build it real quick and upload it, that would be awesome! 

Thank you very much! ❤️

**Comment by prashantgupta24** [comment]
Hi @longgiangp , unfortunately the `robotgo` library on which AMM is based has issues with arm64 builds. Perhaps their latest version supports it, I am using an older version.

FYI I I have been running this on an M1 for quite some time now, rosetta makes it a seamless experience.

**Comment by cnpog** [comment]
@prashantgupta24 I am on M1 mac and I updated all dependencies, after which it worked.
`go get -u ./...`

**Comment by paoloestar** [comment]
good to hear that it works as a native app. Would be great if you could share the build

@prashantgupta24 does this enable you to update your code?

**Comment by prashantgupta24** [comment]
I honestly am not able to find the time to get this done, if someone submits a PR for it, would highly appreciate it!

**Comment by Resousse** [comment]
This PR : https://github.com/prashantgupta24/automatic-mouse-mover/pull/45 should help, because I'm able to compile/use it on my silicon M2 Pro.

**Comment by Resousse** [comment]
@paoloestar , please try the latest release and let me know

**Comment by nighthawk469** [comment]
It may work building from source, but the "amm.x86.zip" in Releases doesn't work on apple silicon, or at least I get this error when I right click>Open "“amm” cannot be opened because the developer cannot be verified.". 

It would be awesome if we could get an apple silicon build in Releases :)

**Comment by prashantgupta24** [comment]
@nighthawk469 the `developer cannot be verified` error is because it's not notarized by apple. If you right click on the `.app` and choose `Open`, it will work. 

That being said, I will get an apple silicon build in the releases soon!

**Comment by nighthawk469** [comment]
I did try right clicking but still I got the same message. I saw this same issue with Alacritty. Something about macos quarantine.
https://github.com/alacritty/alacritty/issues/4673#issuecomment-771291615

**Comment by XanderF** [comment]
I just installed Go for the M2, used master and created amm.app using Make. It worked perfectly. And I haven't had any crashes which I used to get using the amm.app binary on my Intel based Mac.

**Comment by Montrichard** [comment]
@prashantgupta24 any plans to do a Apple Silicon release? thanks

**Comment by yann120** [comment]
I had a warning that this app will no longer be compatible on the next mac os release.
Maybe it would be great to create a build for M1+ macs ?

**Comment by prashantgupta24** [comment]
cc @Resousse is the one supporting it now

**Comment by yann120** [comment]
Yes I found this fork from @Resousse  that works well : https://github.com/Resousse/automatic-mouse-mover
